hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/multipart/S3MultipartUploadCompleteRequest.java

```diff
@@ -175,8 +175,10 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
             failureMessage(requestedVolume, requestedBucket, keyName),
             OMException.ResultCodes.NO_SUCH_MULTIPART_UPLOAD_ERROR);
       }
-      TreeMap<Integer, PartKeyInfo> partKeyInfoMap =
-          multipartKeyInfo.getPartKeyInfoMap();
+
+      TreeMap<Integer, PartKeyInfo> partKeyInfoMap = new TreeMap<>();
+      multipartKeyInfo.getPartKeyInfoMap().forEach(partKeyInfo ->
+          partKeyInfoMap.put(partKeyInfo.getPartNumber(), partKeyInfo));
```

Comment:
Please don't copy to a TreeMap for read only accesses.  We should keep passing `PartKeyInfoMap`.

Reply:
done